### PR TITLE
DOC-10666 Add optional offset clause to delete

### PR DESCRIPTION
**Pull request by lamagnifica**
I set up a preview site for the [updated Delete topic ](https://preview.docs-test.couchbase.com/offset/server/current/n1ql/n1ql-language-reference/delete.html)with [credentials](https://couchbasecloud.atlassian.net/wiki/spaces/DOCS/pages/1971224949/Docs+Team+demo+site+passwords).

[DOC-10666](https://issues.couchbase.com/browse/DOC-10666) adds the ability to include an optional OFFSET clause in DELETE statements.

This PR updates the statement description and syntax, and added another example (#6) and its results. I produced the example and tested using the server:7.6.1-3094 image.

### CODE REVIEW

**Review by julia-cb — APPROVED** [review]

**Comment by julia-cb** [comment]
LGTM!! :)

**Review by dhanyagowrish — APPROVED** [review]

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc [hunk 1]
```diff
@@ -249,4 +269,41 @@ include::example$dml/delete-filter.n1ql[]
...
+====
+This example deletes a subset of the airlines with a country of "France'.
+First, you query to get a list of the airlines in France.
+
+[source,sqlpp]
+----
+include::example$dml/delete-offset-before.n1ql[]
+----
+
+There are 21 documents in this collection with `country="France"`.
```
Comment (lamagnifica) [inline modules/n1ql/pages/n1ql-language-reference/delete.adoc:284]: @simon-dew I did not include the SELECT id FROM... query that I used here or below, is it ok to assume?

File: modules/n1ql/examples/dml/delete-offset-before.jsonc [hunk 2]
```diff
@@ -0,0 +1,22 @@
+id
```
Comment (dhanyagowrish) [inline modules/n1ql/examples/dml/delete-offset-before.jsonc]: A bit of a nit - the "Results" do not match the format of the results returned by its corresponding example query.

File: modules/n1ql/examples/dml/delete-offset.jsonc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+id
```
Comment (dhanyagowrish) [inline modules/n1ql/examples/dml/delete-offset.jsonc]: A bit of a nit - the "Results" do not match the the results returned by its corresponding example query. There are no "ids" returned by the corresponding DELETE query.

**Review by dhanyagowrish — CHANGES_REQUESTED** [review]

File: modules/n1ql/examples/dml/delete-offset-before.jsonc (hunk 2 above)
Comment (lamagnifica) [reply modules/n1ql/examples/dml/delete-offset-before.jsonc]: Agreed, though that is what gets copied when I use the copy icon and paste in -- it is odd that the format is different, but it certainly takes up less space... 
what would be your preference here (and below), @simon-dew ?

File: modules/n1ql/examples/dml/delete-offset.jsonc (hunk 3 above)
Comment (lamagnifica) [reply modules/n1ql/examples/dml/delete-offset.jsonc:1]: that's right, per my question above to Simon @dhanyagowrish I did not include the SELECT id FROM... statements that I used here and above, not sure how detailed these examples tend to be, happy to add/reformat if that is the standard

Comment (lamagnifica) [reply modules/n1ql/examples/dml/delete-offset.jsonc]: I reworded to clarify that this is the Table view, not JSON. 2x

**Review by dhanyagowrish — APPROVED** [review]

File: modules/n1ql/examples/dml/delete-offset-before.jsonc (hunk 2 above)
Comment (simon-dew) [reply modules/n1ql/examples/dml/delete-offset-before.jsonc]: Sorry Alison, all of the other examples on this page (and almost the entire SQL++ reference) show the results in JSON format. Please redo these results in JSON format by select all → copying and pasting from the results pane in the Query Workbench.

If the Copy icon in the Query Workbench only copies the text results, then that is perhaps a UI bug in the workbench.

File: modules/n1ql/examples/dml/delete-offset.jsonc (hunk 3 above)
Comment (simon-dew) [reply modules/n1ql/examples/dml/delete-offset.jsonc]: As before, please redo these results in JSON format by select all → copying and pasting from the results pane in the Query Workbench.

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc [hunk 4]
```diff
@@ -3,7 +3,7 @@
 :page-topic-type: reference
 :imagesdir: ../../assets/images
 
-:authorization-overview: xref:learn:security/authorization-overview.adoc
+:authorization-overview: xref:server:learn:security/authorization-overview.adoc
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/delete.adoc]: I can see why you've done this, but I'm not sure if this change is necessary. As this is the release/7.6 branch, we're already in the server component. We would add the server component to this xref when this change is merged into Capella. Not going to insist that you revert this but it's got me thinking 🤔

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc [hunk 5]
```diff
@@ -172,7 +192,7 @@ include::ROOT:partial$query-context.adoc[tag=section]
 
 WARNING: Please note that running the following examples will permanently delete your sample data.
 To restore your sample data, remove and reinstall the `travel-sample` bucket.
-Refer to xref:manage:manage-settings/install-sample-buckets.adoc[Sample Buckets] for details. 
+Refer to xref:server:manage:manage-settings/install-sample-buckets.adoc[Sample Buckets] for details.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/delete.adoc]: In this case I am sure this change is unnecessary. When we merge this change into Capella, we would have to update this xref anyway, to point to the Capella instructions for installing example files. So adding the server component here is pointless.

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc [hunk 6]
```diff
@@ -249,4 +269,44 @@ include::example$dml/delete-filter.n1ql[]
...
+There are now 11 documents in this collection with `country="France"`, shown for this example in Table view.
+
+.Results
+[source,text]
+----
+include::example$dml/delete-offset.jsonc[]
+----
+
+<.> Documents with the first 10 ids&mdash;the offset&mdash;remain in the airline collection.
+<.> After deleting 10 documents&mdash;the limit&mdash;1 more document remains in the collection.
```
Comment (simon-dew) [inline modules/n1ql/pages/n1ql-language-reference/delete.adoc]: Spaces on either side of em dash? I foresee a style guide discussion in the future.

Also, while `&mdash;` works, you can use the more idiomatic `--` in AsciiDoc.

```suggestion
<.> Documents with the first 10 ids -- the offset -- remain in the airline collection.
<.> After deleting 10 documents -- the limit -- 1 more document remains in the collection.
```

**Review by simon-dew — CHANGES_REQUESTED** [review]
Brilliant, thanks @lamagnifica . Just some minor changes, particularly the results of the example queries.

File: modules/n1ql/examples/dml/delete-offset-before.jsonc (hunk 2 above)
Comment (lamagnifica) [reply modules/n1ql/examples/dml/delete-offset-before.jsonc]: after dropping travel-sample and reinstalling it, I have redone both sets of results

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc (hunk 4 above)
Comment (lamagnifica) [reply modules/n1ql/pages/n1ql-language-reference/delete.adoc]: links are broken in a local build without, I won't test them going forward

File: modules/n1ql/pages/n1ql-language-reference/delete.adoc (hunk 6 above)
Comment (lamagnifica) [reply modules/n1ql/pages/n1ql-language-reference/delete.adoc]: no spaces, -- per Slack convo

**Review by dhanyagowrish — APPROVED** [review]

**Review by simon-dew — APPROVED** [review]
LGTM 👍

**Pull request merged**